src/main/java/it/pagopa/pn/deliverypush/action/notificationview/NotificationViewLegalFactCreationResponseHandler.java

```diff
@@ -0,0 +1,186 @@
+import it.pagopa.pn.deliverypush.action.utils.TimelineUtils;
+import it.pagopa.pn.deliverypush.dto.ext.delivery.notification.NotificationInt;
+import it.pagopa.pn.deliverypush.dto.legalfacts.LegalFactCategoryInt;
+import it.pagopa.pn.deliverypush.dto.mandate.DelegateInfoInt;
+import it.pagopa.pn.deliverypush.dto.radd.RaddInfo;
+import it.pagopa.pn.deliverypush.dto.timeline.TimelineElementInternal;
+import it.pagopa.pn.deliverypush.dto.timeline.details.NotificationViewedCreationRequestDetailsInt;
+import it.pagopa.pn.deliverypush.service.ConfidentialInformationService;
+import it.pagopa.pn.deliverypush.service.NotificationService;
+import it.pagopa.pn.deliverypush.service.PaperNotificationFailedService;
+import it.pagopa.pn.deliverypush.service.TimelineService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+import java.time.Instant;
+import java.util.Optional;
+
+import static it.pagopa.pn.deliverypush.exceptions.PnDeliveryPushExceptionCodes.ERROR_CODE_DELIVERYPUSH_TIMELINENOTFOUND;
+
+@Component
+@AllArgsConstructor
+@Slf4j
+public class NotificationViewLegalFactCreationResponseHandler {
+    private final NotificationService notificationService;
+    private final TimelineService timelineService;
+    private final NotificationCost notificationCost;
+    private final PaperNotificationFailedService paperNotificationFailedService;
+    private final ConfidentialInformationService confidentialInformationService;
+    private final NotificationUtils notificationUtils;
+    private final TimelineUtils timelineUtils;
+    
+    public void handleLegalFactCreationResponse(String iun, int recIndex, DocumentCreationResponseActionDetails actionDetails) {
+        log.info("Start handleLegalFactCreationResponse process - iun={} legalFactId={}", iun, actionDetails.getKey());
+        PnAuditLogEvent recipientAccessLegalFactAuditLog = getAuditLogLegalFact(iun, recIndex, actionDetails);
+        recipientAccessLegalFactAuditLog.log();
+
+        PnAuditLogEvent notificationViewAuditlog = null;
+        
+        try {
+            NotificationInt notification = notificationService.getNotificationByIun(iun);
+
+            Optional<NotificationViewedCreationRequestDetailsInt> notificationViewLegalFactCreationDetailsOpt = timelineService.getTimelineElementDetails(iun, actionDetails.getTimelineId(), NotificationViewedCreationRequestDetailsInt.class);
+
+            if(notificationViewLegalFactCreationDetailsOpt.isPresent()){
+
+                NotificationViewedCreationRequestDetailsInt timelineDetails = notificationViewLegalFactCreationDetailsOpt.get();
+
+                RaddInfo raddInfo = getRaddInfo(timelineDetails);
+                
+                notificationViewAuditlog = generateAuditLog(iun, recIndex, raddInfo, timelineDetails.getDelegateInfo());
```

Comment:
riga 61-74, refactor in un nuovo metodo?
in modo che il metodo chiamante (Questo) scrive l'auditlog di recipientAccessLegalFact
e il nuovo metodo invece si occupa del notificationViewAuditLog?